doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,608 @@
+
+**4.**  Add member vlan(s) in multiple commas or range list, all flag & except flag can be used as well.
+
+**5.**  Add the vlan member(s) in the vlan member table. 
+
+
+__*NOTE: This works same for Adding Multiple VLANs members on a PortChannel*__
+
+
+### Delete Multiple VLAN member on Trunk Port
+
+![Delete-Vlan-Member](https://user-images.githubusercontent.com/61490193/223199172-0ae75308-6e93-4176-9db6-3bcfbd02b948.png)
+
+
+__*Figure 6: Sequence Diagram for deleting VLAN member on Trunk Port*__
+
+**1.** The SONiC CLI will call the vlan_member_input_parser which will parse multiple vlan(s) in a list.
+
+**2.** Check if the members exist in the range .
+
+**3.** Read the existing vlan table and delete vlans(s).
+
+
+__*NOTE: This works same for Delete Multiple VLANs members from a PortChannel*__
+
+## CLI & YANG Model Configuration Commands & Usage 
+
+### CLI Configuration Commands
+
+**1.** config switchport mode <routed|access|trunk> <member_portname>/<member_portchannel>
+ 
+**2.** config vlan add/del -m <comma separated list, range> <vlan_ids>
+
+**3.** config vlan  member add/del -m <all,except, comma separated list, range> <vlan_ids> <member_portname>/<member_portchannel>
+
+
+### YANG Model Configuration 
+
+For Mode attribute, a new type is defined in YANG Model for adding support of "mode" in PORT_TABLE & PORTCHANNEL_TABLE.
+
+#### YANG New type for PORT_TABLE & PORTCHANNEL_TABLE
+
+    typedef switchport_mode {
+        type string {
+            pattern "routed|access|trunk";
+                    }
+        description
+            "SwitchPort Modes for Port & PortChannel";
+                         }
+
+#### YANG Leaf for PORT_TABLE & PORTCHANNEL_TABLE
+
+         leaf mode {
+		description "SwitchPort Modes possible values are routed|access|trunk. Default val is routed. "; 
+		type stypes:switchport_mode; 
+		default "routed";
+				}
+## Config DB Enhancements
+We have added a new “mode” field in PORT & PORT CHANNEL Table .
+
+    "PORT":
+      {
+        "Ethernet0": {
+            "admin_status": "up",
+            "alias": "fortyGigE0/0",
+            "index": "0",
+            "lanes": "25,26,27,28",
+            "mode": "access",
+            "mtu": "9100",
+            "speed": "40000"
+        }
+      }
+  **Assumption: Etherent0 has been configured as access by user
+
+     "PORTCHANNEL": {
+        "PortChannel0001": {
+            "admin_status": "up",
+            "fast_rate": "false",
+            "lacp_key": "auto",
+            "min_links": "1",
+            "mode": "trunk",
+            "mtu": "9100"
+        }
+      }
+ **Assumption: PortChannel0001 has been configured as trunk by user
+
+
+## DB Migrator Enhancements
+As there is a change in schema for PORT & PORT CHANNEL Table in config DB . We have added support to migrate the entries to the new schema in db_migrator.py. A new “mode” field is added whose default value is “trunk” to map it with the old configurations for backward compatibility.
+
```

Comment:
You also need to check if there are any VLAN membership.  If there are no vlan membership the default should be routed instead of Trunk?

Reply:
The mode attribute will appear only if VLAN is configured in the old configuration.  For db_migrator, if there is no VLAN membership it does not explicitly put the mode attribute in the PORT Table, and if we run the command "show interfaces status" it will show it as routed. The default mode for old vlan configurations is set as "trunk" as per the community suggestion/comments. But if you want, we can change the db_migrator behavior to set mode as routed in case of no VLAN membership.

Reply:
If no VLAN is configured, no "mode" should be set.  that is the desired default  DB migrator behavior.  Your previous sections seems to indicate every interface will have a "mode" regardless of being L2 or L3... which I disagree...

Reply:
Agree on this, we will update it.

Reply:
Updated as suggested.